src/mui/input/ReferenceArrayInput.spec.js

```diff
@@ -2,202 +2,704 @@ import React from 'react';
 import assert from 'assert';
 import { shallow } from 'enzyme';
 import sinon from 'sinon';
-import { ReferenceArrayInput } from './ReferenceArrayInput';
-
-describe('<ReferenceArrayInput />', () => {
-    const defaultProps = {
-        crudGetMatching: () => true,
-        crudGetMany: () => true,
-        input: {},
-        reference: 'tags',
-        resource: 'posts',
-        source: 'tag_ids',
-    };
-    const MyComponent = () => <span id="mycomponent" />;
-
-    it('should not render anything if there is no referenceRecord and allowEmpty is false', () => {
-        const wrapper = shallow(
-            <ReferenceArrayInput {...defaultProps}>
-                <MyComponent />
-            </ReferenceArrayInput>
-        );
-        const MyComponentElement = wrapper.find('MyComponent');
-        assert.equal(MyComponentElement.length, 0);
-    });
+import {
+    getSelectedReferencesStatus,
```

Comment:
It feels strange to test individual methods of a component. IMO, these are logic methods which should be placed out of the component. But we'll do that in another PR.